### ISSUE TITLE
slate sniffer & fuzzer setup

### ISSUE THREAD

**Issue by geonno** [quarkslab/starlink-tools#5]
I'm trying to follow up with your slate-sniffer and fuzzer but got stuck with some issues.

1. about the dish-setup
I couldn't find a way to access my dish emulator through ssh connection. what I've tried is this.
    1) I've editted ``rootfs/etc/ssh/ssh_config`` file to enable the password authentication. But when I start the emulator, the edited version doesn't seem to be applied. 
    2) I've tried to just execute the ``dish-setup.sh``file. But it came out with this error. ``ssh: Could not resolve hostname starlink: Temporary failure in name resolution``. 
    3) I've tried to create a ssh-key in the emulator with ``ssh-keygen`` but failed because the filesystem is ``read-only``.

2. about ``generate_service_json.py`` file
at first, I've tried using the ``service_directory`` file for the ``generate_service_json.py``. But my ``service_directory`` file contained only small bits of data which indicated what file to include.  this is my ``service_directory``file.
```
# The UT version of this file only includes the minimal set of entries

%include    service_directory_common_sat_gw_ut.include
%include    service_directory_starlink_slate_sharing.include

################################################################################
# Reliable Telemetry
tron_telemetry_at_pop_grpc                 tron-telemetry-at-pop.svc.starlink.com   8012 tcp_proto

network.pop                                pop.starlink.com                         4754  udp_proto
network.cplane_server                      cplane.starlink.com                      4754  udp_proto
```
so what I did was instead of using ``service_directory`` file, I used the one above(service_directory_starlink_slate_sharing.include) instead and it successfully generated ``service_directory.json``. Is what I did the right way?

I wish to here specific details on how to access the emulator through ssh connection and the comment about what I've done. thank you.

**Comment by CarloRamponi** [comment]
Sorry, I haven't had the time to look at your issue yet.
Have you solved it on your own?

**Comment by geonno** [comment]
yes! I solved it by adding some interfaces shown in your blog. But another problem is going on right now. I am trying to sniff some packets with slate sniffer but no packets seem to be capturing. I think it is because my ``user_terminal_control`` binary is not executing well. when I execute the binary it prints out the following errors.
```
[root@user1 bin]# ./user_terminal_control
external/rocket/src/bullwinkle/all/Configs.cc:43: SacIfNot
external/rocket/src/bullwinkle/all/Configs.cc:45: Error: 'dat' is not a directory.
external/rocket/src/bullwinkle/all/StandardMultiload.cc:128: SacAbortIfNot(,)
external/rocket/src/bullwinkle/all/bin_util.cc:252: SacAbortIfNot(,)
external/rocket/src/bullwinkle/all/bin_util.cc:297: SacAbortIfNot(,)
external/rocket/src/flight/sat/all/user/user_terminal_control_bin.cc:147: SacAbortIfNot(,)
```
what should be the reason?

**Comment by CarloRamponi** [comment]
The first error is due to the working directory in which you are executing the process: it should be `/sx/local/runtime`.
But then you will experience many other errors due to missing hardware, such as the secure element.
To solve those, you'll have to either emulate in QEMU the missing part (in that case, please share your work!) or patch the binary in order not to fail, but the latter is obviously not the optimal solution.

**Comment by geonno** [comment]
Yeah I think I'll have to stick to stick to the QEMU. In your case, how did you get user_terminal_control to work? Could you share your specific works? Thanks

**Comment by CarloRamponi** [comment]
TL;DR: I didn't.

I tried to patch the binary in some places where the missing hardware was required. But at some point, it didn't make any sense because I was basically skipping most of the main functionalities of the program.
Due to time constraints, I couldn't implement the emulation of the hardware in QEMU.

**Comment by geonno** [comment]
Oh I see. But then without the user_terminal_control binary working, how could you capture and do the fuzzing in emulator environment? Is the user_terminal_control binary not necessary for fuzzing and slate sniffer part?

**Comment by CarloRamponi** [comment]
You can still sniff and fuzz other processes but yeah, having the control process working would have been useful

**Comment by geonno** [comment]
Thank you! I think I'll have to try :)
